core-common/src/main/java/eu/arrowhead/common/verifier/CommonNamePartVerifier.java

```diff
@@ -29,7 +29,7 @@
 	//=================================================================================================
 	// members
 	
-	public static final String COMMON_NAME_PART_PATTERN_STRING = "^[A-Za-z](?:[0-9A-Za-z-]*[0-9A-Za-z])?$";
+	public static final String COMMON_NAME_PART_PATTERN_STRING = "^[A-Za-z](?:[0-9A-Za-z-]{1,61})?[0-9A-Za-z]$";
```

Comment:
this one forces the minimum 3 length. I'd use the old one, and in the code before the regexp match I'd check that the length of the candidate variable is <= thant 63 (which should be a constant)